### DISCUSSION TITLE
Error: Connect Econnrefused 127.0.0.1:3092

### DISCUSSION THREAD

**Discussion by n-e-t-d-i-v-e-r** [Ride-The-Lightning/c-lightning-REST#167] [Q&A]
<img width="1254" alt="Screenshot 2023-04-13 at 16 17 28" src="https://user-images.githubusercontent.com/87816768/231787897-d487005b-e7de-4f98-8843-662761e8284c.png">
hi there

i just installed core lightning and RTL. i used for that this manual:
https://raspibolt.org/guide/bonus/lightning/cln.html
https://raspibolt.org/guide/lightning/web-app.html

so far everything is OK. but if i run rtl, i get this:

`ERROR: GetInfo => Get Info Error: {"name":"RequestError","message":"Error: connect ECONNREFUSED 127.0.0.1:3092","cause":{"errno":-111,"code":"ECONNREFUSED","syscall":"connect","address":"127.0.0.1","port":3092},"error":{"errno":-111,"code":"ECONNREFUSED","syscall":"connect","address":"127.0.0.1","port":3092},"options":{"url":"http://127.0.0.1:3092/v1/getinfo","rejectUnauthorized":false,"json":true,"headers":{},"method":"GET","qs":{},"simple":true,"resolveWithFullResponse":false,"transform2xxOnly":false}}
Apr 13 15:07:22 crownero node[4603]: [13/04/2023, 15:07:22] ERROR: CLWebSocket => Web socket error: connect ECONNREFUSED 127.0.0.1:3092`

seems something is wrong with the server config and ports, but i have no idea where and how to look and fix this issue.

may somebody give me a hint here?

**Comment by ShahanaFarooqui** [comment]
Can you please share your rtl-config.json after masking sensitive information?

**Comment by ShahanaFarooqui** [comment]
According to your config, you are serving your RTL server should be serving on port 8080 not on 3092.

**Comment by n-e-t-d-i-v-e-r** [comment]
<img width="375" alt="Screenshot 2023-04-14 at 16 51 24" src="https://user-images.githubusercontent.com/87816768/232078900-86b2bc0d-04ec-4480-91fd-e5d3816e5378.png">
thx for the hint. seems there was an issue with 3092. but now i get this:

`
ERROR: RTL => http://localhost:8080 is already in use.
rtl.service: Main process exited, code=exited, status=1/FAILURE
rtl.service: Failed with result 'exit-code'.
 rtl.service: Consumed 2.204s CPU time.
`

is there a way to change port 8080 to another? if not, where can i see which service actually uses port 8080?

**Reply by ShahanaFarooqui** [reply]
Try ‘lsof -i | grep LISTEN’ to check service ports.

**Comment by n-e-t-d-i-v-e-r** [comment]
holy moly. i changed the port to 9090 and rtl is now up and running. but i geht this now ...
<img width="994" alt="Screenshot 2023-04-14 at 17 11 42" src="https://user-images.githubusercontent.com/87816768/232083884-50495e4f-17a4-48a5-b6fa-30ef5a0ce057.png">

**Reply by ShahanaFarooqui** [reply]
How do you know that RTL is running on 9090? Did you get the message in the log?

If rtl is running on 9090, why your safari is trying to open it at 4001? Is your proxy setup working?

**Comment by n-e-t-d-i-v-e-r** [comment]
rtl looks like this now:
`Started Ride the Lightning.
INFO: RTL => Server is up and running, please open the UI at http://localhost:9090 or your proxy configured url.
INFO: RTL => Server is up and running, please open the UI at http://localhost:9090 or your proxy configured url.
`

rtl-config looks like this:
`  "port": "9090",
  "SSO": {
    "rtlSSO": 0,
    "rtlCookiePath": "",
    "logoutRedirectLink": ""
  },
  "nodes": [
    {
      "index": 1,
      "lnNode": "CLN Node",
      "lnImplementation": "CLN",
      "Authentication": {
        "macaroonPath": "/home/rtl",
        "configPath": "/data/lightningd/config"
      },
      "Settings": {
        "userPersona": "OPERATOR",
        "themeMode": "DAY",
        "themeColor": "INDIGO",
        "fiatConversion": true,
        "currencyUnit": "EUR",
        "logLevel": "ERROR",
        "lnServerUrl": "http://127.0.0.1:3092",
        "enableOffers": true
      }
    }
  ],
  "defaultNodeIndex": 1,
`

torrc looks like this:
`HiddenServiceDir /var/lib/tor/hidden_service_cln_rest/
HiddenServiceVersion 3
HiddenServicePort 9090 127.0.0.1:3092`

and server config looks like this:
`upstream rtl {
  server 127.0.0.1:3000;
}
server {
  listen 4001 ssl;
  proxy_pass rtl;
}
`

Port 8080 is in use from LND

**Reply by ShahanaFarooqui** [reply]
Your CL REST is running on 3092 and you are using 9090 as it’s proxy first.

then you are configuring RTL server on 9090 as well in rtl-config. And in your proxy setup RTL’s port is 3000?

Please note that RTL and cl-rest both are separate servers and need different server ports.

**Comment by n-e-t-d-i-v-e-r** [comment]
and at least that means? 

should i use 9090 for RTL server and RTL’s port?

in the original manual the config looks like this:

nginx:

`upstream rtl {
  server 127.0.0.1:3000;
}
server {
  listen 4001 ssl;
  proxy_pass rtl;
}`

.lightning:

`# network
proxy=127.0.0.1:9050
bind-addr=0.0.0.0:9736
addr=statictor:127.0.0.1:9051/torport=9736
always-use-proxy=true`

config:
`# cln-rest-plugin
plugin=/data/lightningd-plugins-available/c-lightning-REST-0.10.1/clrest.js
rest-port=3092
rest-docport=4091
rest-protocol=http`

cl-rest-config.json:

`{
  "PORT": 3092,
  "DOCPORT": 4091,
  "PROTOCOL": "http",
  "EXECMODE": "production",
  "RPCCOMMANDS": ["*"],
  "DOMAIN": "localhost"
}`

tor:

`############### This section is just for location-hidden services ###
HiddenServiceDir /var/lib/tor/hidden_service_cln_rest/
HiddenServiceVersion 3
HiddenServicePort 8080 127.0.0.1:3092`

but that didn’t worked because of two services uses 8080 as ports.

should i go back to this config? but what and where should i change for not using port 8080 with two services?
otherwise if nothing works i could deinstall LND, which using also port 8080 and have a look after if everything’s fine. but worst case then I’ve got two not working lightning-services on my node ... 😂

sry for stress u with all this questions. i’m good in copy paste, but not in coding and debuging ...

**Reply by saubyk** [reply]
> nginx:
`upstream rtl { server 127.0.0.1:3000; } server { listen 4001 ssl; proxy_pass rtl; }`

Hi @n-e-t-d-i-v-e-r please make the below change and try again:
`upstream rtl { server 127.0.0.1:9090; } server { listen 4001 ssl; proxy_pass rtl; }`

**Comment by n-e-t-d-i-v-e-r** [comment]
yes, that was it. thx you both for supporting me. my noce is up and running now!!!
<img width="1246" alt="Screenshot 2023-04-15 at 10 29 54" src="https://user-images.githubusercontent.com/87816768/232200983-b69f0163-9e1c-4d6e-8c3f-33975d5eba02.png">
